### ISSUE TITLE
Provider crash on KV Access Policy

### ISSUE THREAD

**Issue by favoretti** [hashicorp/terraform-provider-azurerm#11725]
```
Stack trace from the terraform-provider-azurerm_v2.59.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x56461f9]

goroutine 193 [running]:
github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/keyvault.resourceKeyVaultAccessPolicyCreateOrDelete(0xc0004dab60, 0x5fb4b00, 0xc0003cc900, 0x6773177, 0x6, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/keyvault/key_vault_access_policy_resource.go:183 +0x1ff9
github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/keyvault.resourceKeyVaultAccessPolicyDelete(0xc0004dab60, 0x5fb4b00, 0xc0003cc900, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/keyvault/key_vault_access_policy_resource.go:275 +0x54
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc000f15860, 0xc000121220, 0xc000742440, 0x5fb4b00, 0xc0003cc900, 0x100df01, 0xc0018370e0, 0x6334900)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/resource.go:293 +0x4a6
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc000069000, 0xc001831a38, 0xc000121220, 0xc000742440, 0xa261d20, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/provider.go:294 +0x99
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc00000f580, 0x6e77430, 0xc0016b2180, 0xc0004da000, 0xc00000f580, 0xc0016b2180, 0xc000629ba0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin/grpc_provider.go:895 +0x8a5
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x648ccc0, 0xc00000f580, 0x6e77430, 0xc0016b2180, 0xc0016860c0, 0x0, 0x6e77430, 0xc0016b2180, 0xc001808000, 0x2fd)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5/tfplugin5.pb.go:3305 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00023ea80, 0x6ebc1d8, 0xc000702c00, 0xc001806000, 0xc0014fc960, 0xa21f760, 0x0, 0x0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1194 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc00023ea80, 0x6ebc1d8, 0xc000702c00, 0xc001806000, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1517 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0006de8a0, 0xc00023ea80, 0x6ebc1d8, 0xc000702c00, 0xc001806000)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:859 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:857 +0x1fd

Error: The terraform-provider-azurerm_v2.59.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

```

**Comment by KarimJedda** [comment]
Same issue here, upgrading to 1.0.3 didn't fix it.

**Comment by favoretti** [comment]
@KarimJedda this has actually been fixed eons ago :) What do you mean by `1.0.3`? Current version of azurerm provider is `2.70.0`.

**Comment by katbyte** [comment]
hey @favoretti  & @KarimJedda I'm going to close this as @favoretti has mentioned its been fixed by v2.70 & this was a problem in the 2.x version of the provider which is no longer actively supported. 

If this is still an issue with the 3.x version of the provider please do let us know, thanks!
